### PR TITLE
Simple fixes for the diagnostics command

### LINKED ISSUE #81: elastic-agent diagnostics log names on windows
When a diagnostic bundle is created on a windows instances the windows filepath separators are used (in the log sub dirs):

```
Archive:  dia.zip
   creating: meta/
  inflating: meta/elastic-agent-version.yaml
  inflating: meta/metricbeat_monitoring-default.yaml
  inflating: meta/metricbeat-default.yaml
  inflating: meta/filebeat-default.yaml
  inflating: meta/filebeat_monitoring-default.yaml
   creating: config/
  inflating: config/elastic-agent-local.yaml
  inflating: config/elastic-agent-policy.yaml
   creating: logs/
   creating: logs/default/
  inflating: logs/default\filebeat-json.log
  inflating: logs/default\filebeat-json.log-2022-01-27-15-274
  inflating: logs/default\filebeat-json.log-2022-01-27-15-275
  inflating: logs/default\filebeat-json.log-2022-01-27-15-276
  inflating: logs/default\filebeat-json.log-2022-01-27-15-277
  inflating: logs/default\filebeat-json.log-2022-01-27-15-278
  inflating: logs/default\filebeat-json.log-2022-01-27-15-279
  inflating: logs/default\filebeat-json.log-2022-01-27-15-280
  inflating: logs/elastic-agent-json.log-20220105111633
  inflating: logs/elastic-agent-json.log-20220105111732
  inflating: logs/elastic-agent-json.log-20220118162152
  inflating: logs/elastic-agent-json.log-20220118162219
  inflating: logs/elastic-agent-json.log-20220119122234
  inflating: logs/elastic-agent-json.log-20220127154306
```

### LINKED ISSUE #92: Allow elastic-agent diagnostics collect to ignore failures
**Describe the enhancement:**

The `elastic-agent diagnostics collect` command will not produce an archive if it encounters a failure gathering logs or config data (and possible meta-data).
The `collect` command should be able to make a note of these errors in the archive (and echo them to stderr) and continue attempting to gather more data.

**Describe a specific use case for the enhancement or feature:**

When gathering diagnostics information it is preferable to gather an incomplete archive then no archive.

### LINKED ISSUE #190: Diagnostics should include agent ID
When running `elastic-agent diagnostics collect` the Agent's ID should be included somewhere in the report. This is helpful for SDHs to help users with identifying data coming from this given Agent. However, today (I'm using Agent 8.1.0) the `config/elastic-agent-local.yaml` file has an empty string for the agent.id field, for example:

```yaml
fleet:
  enabled: true
  access_api_key: XXXXX
  protocol: http
  host: XXXXX
  hosts:
  - XXXXX
  ssl:
    verification_mode: full
    renegotiation: never
  timeout: 10m0s
  reporting:
    threshold: 10000
    check_frequency_sec: 30
  agent:
    id: "" # this is empty!
```

### PR DESCRIPTION
**Pull request by michel-laterman**
## What does this PR do?

Three separate small fixes for the `elastic-agent diagnostic` command (including `collect`):

1. `diagnostics collect` will ignore failures where possible and log to stderr and to an `errors.txt` file in the archive.
2. Fixed log names when `diagnostics collect` is ran on a Windows machine
3. Add the elastic-agent ID to the output of `elastic-agent diagnostics` and into the archive in the `meta/elastic-agent.yaml` and `config/elastic-agent-local.yaml` files

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [] ~~I have made corresponding changes to the documentation~~
- [] ~~I have made corresponding change to the default configuration files~~
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

### Testing log name fix

Run the agent on a Windows machine and run `elastic-agent diagnostics collect`.
Extract the archive (on a *nix machine) and ensure that the logs are named correctly, i.e.:
```
$ unzip elastic-agent-diagnostics-2022-03-28T20-26-59Z-00.zip
Archive:  elastic-agent-diagnostics-2022-03-28T20-26-59Z-00.zip
   creating: meta/
  inflating: meta/elastic-agent-version.yaml
  inflating: meta/filebeat-default.yaml
  inflating: meta/metricbeat-default.yaml
  inflating: meta/filebeat_monitoring-default.yaml
  inflating: meta/metricbeat_monitoring-default.yaml
   creating: config/
  inflating: config/elastic-agent-local.yaml
  inflating: config/elastic-agent-policy.yaml
  inflating: config/filebeat_default.yaml
  inflating: config/metricbeat_default.yaml
  inflating: config/fleet_monitoring_default.yaml
   creating: logs/
   creating: logs/default/
  inflating: logs/default/filebeat-20220328-1.ndjson
  inflating: logs/default/filebeat-20220328.ndjson
  inflating: logs/default/metricbeat-20220328-1.ndjson
  inflating: logs/default/metricbeat-20220328.ndjson
  inflating: logs/elastic-agent-20220328-1.ndjson
  inflating: logs/elastic-agent-20220328.ndjson
```

### Agent ID

Run `elastic-agent diagnostics` and ensure `id` is part of the agent description:
```
vagrant@ubuntu-focal:~$ sudo /opt/Elastic/Agent/elastic-agent diagnostics
elastic-agent  id: 814a02ee-0a26-45dd-bf96-e7005260a3d5                version: 8.2.0
               build_commit: cafa251125139303a7795977455a13d97d99095e  build_time: 2022-03-28 23:41:57 +0000 UTC  snapshot_build: true
```

`elastic-agent diagnostics collect` should include this id as part of `meta/elastic-agent-version.yaml` and `config/leastic-agent-local.yaml`.

## Related issues

- Closes #190 
- Closes #81 
- Closes #92

### CODE REVIEW

File: internal/pkg/agent/cmd/diagnostics.go [hunk 1]
```diff
@@ -520,7 +560,9 @@ func zipLogs(zw *zip.Writer) error {
 			return fmt.Errorf("unable to walk log dir: %w", fErr)
 		}
 
-		name := strings.TrimPrefix(path, logPath)
+		// name is the relative dir/file name replacing any filepath seperators with /
+		// this will clean log names on windows machines and will nop on *nix
+		name := filepath.ToSlash(strings.TrimPrefix(path, logPath))
```
Comment (ph) [inline internal/pkg/agent/cmd/diagnostics.go:565]: TIL about `ToSlash`

**Review by ph — APPROVED** [review]
LGTM

**Comment by amolnater-qasource** [comment]
Hi @michel-laterman 
We have revalidated this PR on latest 8.3 Snapshot Kibana cloud environment with Windows Agent.
We had below observations:
On running `elastic-agent diagnostics` and with `collect` from Elastic/Agent folder:
- We observed Agent logs are named correctly.
- Diagnostics command includes: Agent id.
- ID is also available under `meta/elastic-agent-version.yaml` and `config/elastic-agent-local.yaml`.
- No errors.txt is observed in collected zip file.

**Screenshots:**




**Logs:**
[elastic-agent-diagnostics-2022-05-10T09-17-15Z-00.zip](https://github.com/elastic/elastic-agent/files/8660509/elastic-agent-diagnostics-2022-05-10T09-17-15Z-00.zip)


**Build details:**
BUILD: 52512
COMMIT: df225b213b188c81888141cee2ec191424fc0649

## Queries:

- Could you please confirm if we can run this  `.\elastic-agent diagnostics` command from extracted location on Windows?
- While running this command from extracted folder `C:\Program Files\elastic-agent-8.3.0-SNAPSHOT-windows-x86_64` we get few errors and also get error.txt collected in zip file.

Screenshot and logs while running diagnostics command from `C:\Program Files\elastic-agent-8.3.0-SNAPSHOT-windows-x86_64` is attached below:
[elastic-agent-diagnostics-2022-05-10T09-43-19Z-00.zip](https://github.com/elastic/elastic-agent/files/8660552/elastic-agent-diagnostics-2022-05-10T09-43-19Z-00.zip)



Please let us know if we are missing anything.
Thanks

**Comment by michel-laterman** [comment]
I think running the command from the extracted location instead of the installed one leads to the agent attempting to use an incorrect npipe for communications (and getting errors). I think this is expected behaviour.

**Comment by amolnater-qasource** [comment]
Hi @michel-laterman 
Thanks for confirming this as expected behaviour.
As the testing on this is done from our end. We are marking this as QA:Validated.

Thanks

**Pull request merged**